Libraries/LibWeb/CSS/Transformation.cpp

```diff
@@ -25,6 +26,13 @@ ErrorOr<Gfx::FloatMatrix4x4> Transformation::to_matrix(Optional<Painting::Painta
             context.length_resolution_context = Length::ResolutionContext::for_layout_node(paintable_box->layout_node());
 
         return m_values[index].visit(
+            [&](CSS::AngleOrCalculated const& value) -> ErrorOr<float> {
+                if (!value.is_calculated())
+                    return value.value().to_radians();
+                if (value.is_calculated() && value.calculated()->resolves_to_angle())
+                    return value.resolved(context)->to_radians();
+                return Error::from_string_literal("Transform contains non absolute units");
+            },
```

Comment:
The AngleOrCalculated code doesn't actually need to change, does it? `value.resolved(context)` should be enough.

The reason that LengthPercentage does things manually is because percentages require a Node to resolve them as a length, but we don't care about that if we don't have a percentage. (NumberPercentage does things manually just because it's weird.)

Side-note: I understand why you reorganized the code, but it does make it harder to spot the actual change. :sweat_smile: